src/s1reader/s1_orbit.py

```diff
@@ -56,16 +61,65 @@ def download_orbit(safe_file: str, orbit_dir: str):
     # Find precise orbit first
     orbit_dict = get_orbit_dict(mission_id, start_time,
                                 end_time, 'AUX_POEORB')
+
     # If orbit dict is empty, find restituted orbits
     if orbit_dict is None:
         orbit_dict = get_orbit_dict(mission_id, start_time,
                                     end_time, 'AUX_RESORB')
+
     # Download orbit file
-    orbit_file = os.path.join(orbit_dir, f"{orbit_dict['orbit_name']}.EOF")
-    if not os.path.exists(orbit_file):
-        download_orbit_file(orbit_dir, orbit_dict['orbit_url'])
+    if orbit_dict is not None:
+        orbit_file = os.path.join(orbit_dir, f"{orbit_dict['orbit_name']}.EOF")
+        if not os.path.exists(orbit_file):
+            download_orbit_file(orbit_dir, orbit_dict['orbit_url'])
 
-    return orbit_file
+        return orbit_file
+
+    # POEORB is not found, or there is no RESORB file that
+    # covers the sensing period + margin at the starting time.
+    # Try to find two subsequent RESORB files that covers the
+    # sensing period + margins at the starting time.
+    if orbit_dict is None:
+        pad_short = datetime.timedelta(seconds = PADDING_SHORT)
+        print('Attempting to download RESORB files.')
+        orbit_dict_earlier = get_orbit_dict(mission_id,
+                                            start_time,
+                                            start_time + 2 * pad_short, 'AUX_RESORB')
+
+        orbit_dict_later = get_orbit_dict(mission_id,
+                                          start_time + margin_start_time - pad_short,
+                                          end_time + pad_short,
+                                          'AUX_RESORB')
+
+        orbit_dict_list = [orbit_dict_earlier, orbit_dict_later]
+
+        if orbit_dict_list:
+            orbit_file_indv_list = []
+            for orbit_dict in orbit_dict_list:
```

Comment:
In case that we don't find the RESORB pair, the function will send out the warning message and return `None`. It seems like that the overall behavior of this module is to return `None` when there is no orbit file is found, so wanted to let this new feature to behave the same.

Reply:
Right above this if statement, we check if all the elements in `orbit_dic_list` (which is `resorb_dict_list` now) are valid.  If not, we early return. The snippet for that is as below. 

```python
    if not all(resorb_dict_list):
        warnings.warn('Cannot find RESORB pair that covers the timeframe in the query')
        return
```

Therefore all the elements in the list has valid values at this point.